src/python/WMCore/WMRuntime/Scripts/SetupCMSSWPset.py

```diff
@@ -333,32 +331,19 @@ def handleChainedProcessingTweak(self):
         """
         _handleChainedProcessing_
 
-        In order to handle chained processing it's necessary to feed
-        output of one step/task (nomenclature ambiguous) to another.
-        This method creates particular mapping in a working Trivial
-        File Catalog (TFC).
+        When a job has multiple cmsRun steps, we need to tweak the
+        subsequent PSet configuration such that it reads file in
+        from the local disk (job sandbox).
         """
         self.logger.info("Handling chained processing job")
-        # first, create an instance of TrivialFileCatalog to override
-        tfc = TrivialFileCatalog()
+
         # check the jobs input files
-        inputFile = ("../%s/%s.root" % (self.step.data.input.inputStepName,
-                                        self.step.data.input.inputOutputModule))
-        tfc.addMapping("direct", inputFile, inputFile, mapping_type="lfn-to-pfn")
-        tfc.addMapping("direct", inputFile, inputFile, mapping_type="pfn-to-lfn")
+        inputFile = ("file:../%s/%s.root" % (self.step.data.input.inputStepName,
+                                             self.step.data.input.inputOutputModule))
 
         self.tweak.addParameter('process.source.fileNames', "customTypeCms.untracked.vstring(%s)" % [inputFile])
         self.tweak.addParameter("process.maxEvents", "customTypeCms.untracked.PSet(input=cms.untracked.int32(-1))")
-
-        tfcName = "override_catalog.xml"
-        tfcPath = os.path.join(os.getcwd(), tfcName)
-        self.logger.info("Creating override TFC and saving into '%s'", tfcPath)
-        tfcStr = tfc.getXML()
-        with open(tfcPath, 'w') as tfcFile:
-            tfcFile.write(tfcStr)
-
-        self.step.data.application.overrideCatalog = "trivialcatalog_file:" + tfcPath + "?protocol=direct"
-
+        self.logger.info("Chained PSet tweaked with maxEvents: -1, and fileNames: %s", inputFile)
```

Comment:
It would be good to add also the section of the pset config being altered. So the better string would be:
"  process.maxEvents: -1, and process.source.fileNames: %s"

Reply:
I would say it's too much. If we get to that point, we are likely already looking at the source code :)
Thanks for the review Todor.

Reply:
> If we get to that point, we are likely already looking at the source code

If something could be clarified in advance, I wouldn't leave it to be deducted  from the source code during a debugging process, but as you wish. All good for me.